### PR TITLE
e2e: add http route permanet direct cases

### PR DESCRIPTION
**Pull request by whalecold**
### Ⅰ. Describe what this PR did

issue: https://github.com/alibaba/higress/issues/168

### CODE REVIEW

**Comment by whalecold** [comment]
I try to use command `make ingress-conformance-test` to test locally,  but failed with the error  
```
# github.com/alibaba/higress/pkg/ingress/mcp
pkg/ingress/mcp/generator.go:36:30: push.AllServiceEntries undefined (type *model.PushContext has no field or method AllServiceEntries)
```

File: test/ingress/conformance/tests/httproute-permanent-redirect.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+					},
+				},
+				Response: http.AssertionResponse{
+					ExpectedResponse: http.Response{
+						StatusCode: 301,
+					},
+				},
+			},
+		}
+		t.Run("Canary HTTPRoute Traffic Split", func(t *testing.T) {
```
Comment (CH3CHO) [inline test/ingress/conformance/tests/httproute-permanent-redirect.go:54]: "Canary HTTPRoute Traffic Split" doesn't sound right here.

**Comment by Xunzhuo** [comment]
> I try to use command `make ingress-conformance-test` to test locally, but failed with the error
> 
> ```
> # github.com/alibaba/higress/pkg/ingress/mcp
> pkg/ingress/mcp/generator.go:36:30: push.AllServiceEntries undefined (type *model.PushContext has no field or method AllServiceEntries)
> ```

CI passed for now, I think we can merge after resolving https://github.com/alibaba/higress/pull/209#discussion_r1113729050

**Comment by Xunzhuo** [comment]
@whalecold Have you exec `make prebuild` before this?

**Comment by whalecold** [comment]
> @whalecold Have you exec `make prebuild` before this?

Done, PTAL.

**Review by Xunzhuo — APPROVED** [review]

**Comment by Xunzhuo** [comment]
Thank you @whalecold, good job 👍

**Pull request merged**